R/package.R

```diff
@@ -6,40 +6,75 @@
 #' @importFrom pkgdepends remotes
 #' @importFrom pkginstall install_packages
 #' @importFrom crayon blue
+#' @importFrom glue backtick glue
 #' @export
 pkg_install <- function(pkg, lib = .libPaths()[[1L]], num_workers = 1L) {
   r <- remotes$new(pkg, library = lib)
 
   # Solve the dependency graph
   r$solve()
+  sol <- r$get_solution()
+
+  # Solution error?
+  if (sol$status != "OK") {
+    msg <- paste(format(sol$failures), collapse = "\n")
+    cat("\n")
+    stop("Cannot install packages:\n", msg, call. = FALSE)
+  }
 
   # Actually download packages as needed
   r$download_solution()
 
+  # Download errors?
+  dl <- r$get_solution_download()
+  if (any(bad <- tolower(dl$data$download_status) == "failed")) {
+    msgs <- vcapply(
+      which(bad),
+      function(i) {
+        urls <- paste(backtick(dl$data$sources[[i]]), collapse = ", ")
```

Comment:
You can use `glue::collapse(backtick(dl$data$sources[[i]]), ",")` for this if you want...

Reply:
Yeah, I think I'll just write a helper function, `fmt_items()` or sg like that. Both the `paste` and the `glue` one are sub-optimal.